godot-core/src/builtin/array.rs

```diff
@@ -56,10 +56,24 @@ use super::meta::{
 // GodotType`. Whew. This could be fixed by splitting up `GodotFfi` if desired.
 #[repr(C)]
 pub struct Array<T: GodotType> {
+    // Safety Invariant: The type of all values in `opaque` matches the type `T`.
     opaque: sys::types::OpaqueArray,
     _phantom: PhantomData<T>,
 }
 
+/// Guard that can only call immutable methods on the array.
+struct InnerArrayRef<'a> {
+    inner: inner::InnerArray<'a>,
+}
+
+impl<'a> std::ops::Deref for InnerArrayRef<'a> {
+    type Target = inner::InnerArray<'a>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
```

Comment:
While "ref" is often used as a counterpart to "mut", I feel in this context it doesn't express the intent enough, and sounds more like "reference".

Maybe something like `ImmutableInnerArray` or so?

Reply:
The methods could also be called `as_inner` and `as_inner_mut` 🤔